eclipse-theia/theia
# fix #8210: make validation message of existing resources more specific

Signed-off-by: Stefan Winkler <stefan@winklerweb.net>



#### What it does
Fixes #8210

Instead of providing a generic validation message in `validateFileName()`, if the resource exists, we resolve the resource and decide whether it is a folder or not. Depending on the outcome, we use the correct type (file or folder) in the validation message.

#### How to test
I have added respective test cases to the unit tests (-> yarn test).
To verify manually, open Theia with any workspace containg at least a file and a folder.
- Execute the command „New File“ and enter the name of the existing file.
- Execute the command „New Folder“ and enter the name of the existing file.
=> the validation message should say that A file „XXX“ already exists at this location

- Execute the command „New File“ and enter the name of the existing folder.
- Execute the command „New Folder“ and enter the name of the existing folder.
=> the validation message should say that A folder „XXX“ already exists at this location

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
